src/guide.jl

```diff
@@ -492,11 +492,18 @@ end
 
 struct ManualColorKey{C<:Color} <: Gadfly.GuideElement
     title::Union{AbstractString, (Void)}
-    labels::Vector{AbstractString}
-    colors::Vector{C}
+    labels::OrderedDict{C, AbstractString}
+end
+function ManualColorKey(title, labels, colors::Vector{C}) where {C<:Color}
+    labeldict = OrderedDict{C, AbstractString}()
+    for (c, l) in zip(colors, labels)
+        if c in keys(labeldict)
+            error("Colors should not appear more than once")
+        end
```

Comment:
`haskey(labeldict,c) || error()` would be more succint.  otherwise, lgtm once CI passes.